File: Golang/summary_ranges/228.go
package summary_ranges

import "fmt"

func summaryRanges(nums []int) []string {
	result := []string{}

	for i := 0; i < len(nums); i++ {
		end := i
		for end+1 < len(nums) && nums[end+1]-nums[end] == 1 {
			end++
		}

		if end > i {
			result = append(result, fmt.Sprintf("%d->%d", nums[i], nums[end]))
		} else {
			result = append(result, fmt.Sprintf("%d", nums[end]))
		}

		i = end
	}

	return result
}

File: TypeScript/summary_ranges/228.ts
function summaryRanges(nums: number[]): string[] {
  const result: string[] = [];

  for (let i = 0; i < nums.length; i++) {
    let end = i;
    while (end + 1 < nums.length && nums[end + 1] - nums[end] == 1) {
      end++;
    }

    if (end > i) {
      result.push(`${nums[i]}->${nums[end]}`);
    } else {
      result.push(`${nums[end]}`);
    }

    i = end;
  }

  return result;
}

export { summaryRanges };
